src/keyboardhelper.h

```diff
@@ -0,0 +1,332 @@
+#pragma once
+
+// system includes
+#include <futurecpp.h>
+
+// esp-idf includes
+#include <esp_log.h>
+
+// 3rdparty lib includes
+#include <fmt/core.h>
+#include <strutils.h>
+
+// local includes
+#include "display.h"
+#include "screenmanager.h"
+#include "tftinstance.h"
+
+namespace espgui {
+namespace {
+    constexpr const char * const TAG = "KeyboardHelper";
+    // all ascii chars from '!' to '~'
+    constexpr const char * const KEYBOARD_SCREEN_1 =  "\"!$%&/()=?QWERTZUIOPASDFGHJKL YXCVBNM";
+    constexpr const char * const KEYBOARD_SCREEN_2 =   "1234567890qwertzuiopasdfghjkl yxcvbnm";
+    constexpr const char * const KEYBOARD_SCREEN_3 = "#'*+-,.;:_<|>\\@[]^_`{}~\"!$%&/ ()=?+-*";
+    constexpr const char * const SHIFT = "Shift";
+    constexpr const char * const SPACE = "Space";
+    constexpr const char * const BACKSPACE = "Back";
+    constexpr const char * const ENTER = "Enter";
+} // namespace
+
+template<typename TDisplay = espgui::Display>
+class Keyboard
+{
+public:
+    explicit Keyboard(TDisplay &display) : m_display(display) {}
+
+    void start()
+    {
+        const auto isLandscape = espgui::isLandscape();
+        m_keyboard_start_y = isLandscape ? 98 : 120;
+
+        tft.fillRect(1, m_keyboard_start_y - 10, tft.width()-1, tft.height() - m_keyboard_start_y - (isLandscape ? 0 : 30), TFT_BLACK);
+        tft.drawSunkenRect(1, m_keyboard_start_y - 10, tft.width()-1, tft.height() - m_keyboard_start_y - (isLandscape ? 0 : 30), TFT_WHITE, TFT_GREY, TFT_BLACK);
+
+        updateCharLength();
+        drawKeyboard();
+    }
```

Comment:
wenn impl schon im header sein muss, dann zumindest bitte unter der klasse und nicht inline in der klasse. macht zb größe abschätzen unmöglich ohne millionen zeilen zu lesen